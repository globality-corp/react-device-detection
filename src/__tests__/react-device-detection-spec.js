import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import reactDeviceDetection from '../index';

describe('reactDeviceDetection', () => {

    it('should decorate wrapped component by props with requested in HOC config device/browser information', () => {
        const FakeComponent = () => <div>Test component</div>;
        const WrappedComponent = reactDeviceDetection('mobile', 'firefox', 'ie', 'chrome')(FakeComponent);
        const wrapper = shallow(<WrappedComponent otherProp={20} />);
        const tree = toJson(wrapper);

        expect(tree).toMatchSnapshot();
        expect(wrapper.props().isMobile).toBeDefined();
        expect(wrapper.props().isFirefox).toBeDefined();
        expect(wrapper.props().isIE).toBeDefined();
        expect(wrapper.props().isChrome).toBeDefined();
    });

    it('should decorate wrapper component and pass only requested props', () => {
        const FakeComponent = () => <div>Test component</div>;
        const WrappedComponent = reactDeviceDetection('mobile')(FakeComponent);
        const wrapper = shallow(<WrappedComponent otherProp={20} />);
        const tree = toJson(wrapper);

        expect(tree).toMatchSnapshot();
        expect(wrapper.props().isMobile).toBeDefined();
        expect(wrapper.props().isChrome).toBeUndefined();
    });
});
