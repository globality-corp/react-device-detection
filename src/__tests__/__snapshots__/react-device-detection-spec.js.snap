// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reactDeviceDetection should decorate wrapped component by props with requested in HOC config device/browser information 1`] = `
<FakeComponent
  isChrome={false}
  isFirefox={false}
  isIE={false}
  isMobile={false}
  otherProp={20}
/>
`;
